src/components/BibleDropDown/BibleBooks.js

```diff
@@ -1,76 +1,130 @@
 import React, { useContext } from 'react';
 import { makeStyles } from '@material-ui/core/styles';
-import InputLabel from '@material-ui/core/InputLabel';
-import MenuItem from '@material-ui/core/MenuItem';
-import FormControl from '@material-ui/core/FormControl';
-import Select from '@material-ui/core/Select';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Dialog from '@material-ui/core/Dialog';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import { Button, Checkbox, Typography } from '@material-ui/core';
+import Grid from '@material-ui/core/Grid';
+import PropTypes from 'prop-types';
 import { BookContext } from './BookContext';
-import { bibleChapters } from '../../store/bibleData';
 
 const useStyles = makeStyles((theme) => ({
   formControl: {
     margin: theme.spacing(1),
     minWidth: 120,
   },
+  bookButton: {
+    width: 90,
+    padding: '0 8px',
+  },
+  bookCard: {
+    width: 700,
+  },
+  margin: {
+    margin: theme.spacing(1),
+  },
+  checkBox: {
+    padding: 5,
+  },
+  label: {
+    marginRight: 0,
+    width: 70,
+  },
 }));
 
-export default function BibleBooks() {
+export default function BibleBooks(props) {
   const classes = useStyles();
-  const [book, setBook] = React.useState('');
-  const [chapter, setChapter] = React.useState('');
-  const [chapterList, setChapterList] = React.useState([]);
-
   const { books } = useContext(BookContext);
+  const { value, onChange } = props;
+  const [bookList, setBookList] = React.useState(value);
+  const [open, setOpen] = React.useState(false);
 
-  const bookChanged = (event) => {
-    setBook(event.target.value);
-    const bookId = event.currentTarget.getAttribute('data-bookid');
-    const chapters = bibleChapters[bookId];
-    setChapterList(new Array(chapters).fill(0).map((_, i) => i + 1));
+  const closeBookListing = () => {
+    onChange(bookList);
```

Comment:
We can sort the books using default order.
Have sent you some code in teams to do that. you can use that code